### PR TITLE
Adds PiGallery2 - bpatrik.github.io/pigallery2

### PR DESCRIPTION
**Pull request by geongeorge**
### General information on project
This pull request proposes to add a new rock-on for the following project:
- name: PiGallery2
- website: https://bpatrik.github.io/pigallery2/
- description: This is a fast (like faster than your PC fast) directory-first photo gallery website, optimised for running on low resource servers (especially on raspberry pi).

### Information on docker image
- docker image: https://hub.docker.com/r/bpatrik/pigallery2
- is an official docker image available for this project?: Yes

### Checklist
- [x] Passes [JSONlint](https://jsonlint.com) validation
- [x] Entry added to `root.json` in _alphabetical_ order (for new rock-on only)
- [x] `"description"` object lists and links to the docker image used
- [x] `"description"` object provides information on the image's particularities (advantage over another existing rock-on for the same project, for instance)
- [x] `"website"` object links to project's main website

### CODE REVIEW

**Comment by geongeorge** [comment]
Hi @phillxnet I got stuck with the [pull request for Piwigo](https://github.com/rockstor/rockon-registry/pull/327), Found an alternate project.
This container is 100% self-contained. Would be amazing if you can merge after reviewing

File: pigallery2.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+    "PiGallery2": {
+        "containers": {
+            "pigallery2": {
+                "image": "bpatrik/pigallery2",
+                "launch_order": 1,
+                "ports": {
+                    "4001": {
```
Comment (Hooverdan96) [inline pigallery2.json:8]: According to the instruction I found here, it seems that by default there is a mapping to the port 80 for the container.  So this should be set to 80. I think default of 4001 you set is fine. This was the only way I could make this work.
```suggestion
                    "80": {
```

File: pigallery2.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		            },
+                    "PGID": {
+                        "description": "Enter a valid GID to use along with the above UID. It(or the above UID) must have full permissions to all Shares mapped in the previous step.",
+                        "label": "GID to run Ombi as.",
+                        "index": 2
+                    }
+                }
+            }
+        },
+        "description": "A fast directory-first photo gallery website, with rich UI, optimized for running on low resource servers (especially on raspberry pi) <a href='https://hub.docker.com/r/bpatrik/pigallery2' target='_blank'>https://hub.docker.com/r/bpatrik/pigallery2</a>, available for amd64 and arm64 architecture.</p>",
```
Comment (Hooverdan96) [inline pigallery2.json:48]: Small cleanup on the image/architecture portion
```suggestion
       "description": "A fast directory-first photo gallery website, with rich UI, optimized for running on low resource servers (especially on raspberry pi).<p>Based on official docker image: <a href='https://hub.docker.com/r/bpatrik/pigallery2' target='_blank'>https://hub.docker.com/r/bpatrik/pigallery2</a>, available for amd64 and arm64 architecture.</p>",
```

**Comment by Hooverdan96** [comment]
@geongeorge , take a look at my comments up there. On the proposed port change, it was the only way I could make it work, since the mapping should be  0.0.0.0:4001->80/tcp I think.

File: pigallery2.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                    },
+                    "/images": {
+                        "description": "Choose a Share for Media",
+                        "label": "Media Storage"
+                    }
+                },
+                "environment": {
+                    "PUID": {
+                        "description": "Enter a valid UID to run Ombi as. It must have full permissions to all Shares mapped in the previous step.",
+                        "label": "UID to run Ombi as.",
```
Comment (Hooverdan96) [inline pigallery2.json:37]: probably want to change that for this Rockon
```suggestion
                        "label": "UID to run PiGallery2 as.",
```

File: pigallery2.json [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                        "label": "Temp Storage"
+                    },
+                    "/images": {
+                        "description": "Choose a Share for Media",
+                        "label": "Media Storage"
+                    }
+                },
+                "environment": {
+                    "PUID": {
+                        "description": "Enter a valid UID to run Ombi as. It must have full permissions to all Shares mapped in the previous step.",
```
Comment (Hooverdan96) [inline pigallery2.json:36]: probably want to change that for this Rockon
```suggestion
                        "description": "Enter a valid UID to run PiGallery2 as. It must have full permissions to all Shares mapped in the previous step.",
```

**Comment by phillxnet** [comment]
@geongeorge Could I also chip in with a request to re-base on currently master if your would please. This massively helps with us evaluating just prior to publishing. Post what-ever @Hooverdan96 has requested of course.

Thanks. And agreed this would make for a nice addition. Hopefully we are on a new merging spree here now again.

**Comment by phillxnet** [comment]
@geongeorge Thanks for your efforts here. Closing as not response to requested changes. Do please re-open if you are happy to address the requested changes from @Hooverdan96's review and my rebase request.

**Pull request closed without merging**